packages/kit/src/runtime/app/forms.js

```diff
@@ -161,14 +165,28 @@ export function enhance(form_element, submit = () => {}) {
 		let result;
 
 		try {
+			const is_valid_enctype = enctype?.match(
+				/application\/x-www-form-urlencoded|multipart\/form-data|text\/plain/
+			);
+
+			const headers = new Headers({
+				accept: 'application/json',
+				'Content-Type': is_valid_enctype ? enctype : 'application/x-www-form-urlencoded',
+				'x-sveltekit-action': 'true'
+			});
+
+      // @ts-expect-error `URLSearchParams(form_data)` is kosher, but typescript doesn't know that
+			const body = enctype === 'multipart/form-data' ? form_data : new URLSearchParams(form_data);
+
+			if (enctype === 'multipart/form-data' && headers.get('Content-Type')) {
+				headers.delete('Content-Type');
+			}
```

Comment:
so this was actually needed due to serialization of `mutlipart/form-data` requests. if we are using this type of request for our form, explicitly defining the `Content-Type` will prevent the serialization to happen and thus it will break the form's functionality.

But still, to have the code more clear and clean i've rewritten this bit with e22c88c478e247a0b3cf004e057961c034f358c5

Reply:
Thanks for that. I've learned something new again.
For future reference:
> When using FormData to submit POST requests using [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest) or the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) with the multipart/form-data content type (e.g. when uploading files and blobs to the server), do not explicitly set the [Content-Type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type) header on the request. Doing so will prevent the browser from being able to set the Content-Type header with the boundary expression it will use to delimit form fields in the request body.

https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest_API/Using_FormData_Objects#sending_files_using_a_formdata_object